java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -1760,4 +1763,48 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_copyColumnViewToCV(JNIEnv
   }
   CATCH_STD(env, 0)
 }
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_replaceChildrenWithViews(
+    JNIEnv *env, jobject j_object, jlong j_handle, jintArray j_indices, jlongArray j_children) {
+
+  JNI_NULL_CHECK(env, j_handle, "native handle is null", 0);
+  JNI_NULL_CHECK(env, j_indices, "child indices to replace can't be null", 0);
+  JNI_NULL_CHECK(env, j_children, "children to replace can't be null", 0);
+
+  try {
+    cudf::jni::native_jpointerArray<cudf::column_view> children_to_replace(env, j_children);
+    cudf::jni::native_jintArray indices(env, j_indices);
+    JNI_ARG_CHECK(env, indices.size() == children_to_replace.size(), "The indices size and children size should match", 0);
+
+    cudf::column_view *n_col_view = reinterpret_cast<cudf::column_view *>(j_handle);
+    cudf::type_id id = n_col_view->type().id();
+    JNI_ARG_CHECK(env, id == cudf::type_id::STRUCT || id == cudf::type_id::LIST, "Only nested types are allowed", 0);
+    if (id == cudf::type_id::LIST) {
+      JNI_ARG_CHECK(env, children_to_replace.size() == 1, "LIST can only have one child to replace", 0);
+    }
+
+    std::map<int32_t, cudf::column_view*> m;
+    for (int i = 0 ; i < indices.size() ; i++) {
+      m[indices[i]] = children_to_replace[i];
```

Comment:
So if there are duplicate indexes then the last one wins?  We really should document that as well in the docs, or perhaps check for it and throw an exception if we find it.